### PR TITLE
If ConsoleWindow not available, try to find a window handle from the …

### PR DESCRIPTION
**Pull request by TomKuhn**
Try to find a window handle on parent process

### CODE REVIEW

**Comment by TomKuhn** [comment]
Just meant as an example, for people running this from e.g a Windows Forms application and running CredentialProvider.Microsoft.exe with no console window, causing the interactive provider to fail on the broker.

**Comment by TomKuhn** [comment]
@microsoft-github-policy-service agree [company="Microsoft"]

File: src/Authentication/AzureArtifacts.cs [hunk 1]
```diff
@@ -102,7 +104,84 @@ private static IntPtr GetConsoleOrTerminalWindow()
         IntPtr consoleHandle = GetConsoleWindow();
         IntPtr handle = GetAncestor(consoleHandle, GetAncestorFlags.GetRootOwner);
 
-        return handle;
+		if (handle == IntPtr.Zero)
```
Comment (jmyersmsft) [inline src/Authentication/AzureArtifacts.cs:107]: Please indent with 4 spaces and not tabs.

**Comment by embetten** [comment]
Closing - please see comment on issue for more details and discussion https://github.com/microsoft/artifacts-credprovider/issues/472#issuecomment-1967597439 .

**Pull request closed without merging**